iquip/apps/scheduler.py

```diff
@@ -212,17 +245,77 @@ def sizeHint(self, option: QStyleOptionViewItem, index: QModelIndex) -> QSize:
         return experimentView.sizeHint()
 
 
+class SchedulerPostWorker(QObject):
+    """Worker for posting a request to the proxy server, targeting the scheduler.
+
+    Signals:
+        done: The signal is emitted when the procedure of the worker is done.
+
+    Attributes:
+        mode: The type of command that is requested to the server.
+        rid: The run identifier value of the target experiment.
+    """
+    done = pyqtSignal()
+
+    def __init__(self, mode: str, rid: int):
+        """Extended.
+
+        Args:
+            mode: The type of command that is requested to the server.
+            rid: The run identifier value of the target experiment.
+        """
+        super().__init__()
+        self.mode = mode
+        self.rid = rid
+
+    def run(self):
+        """Overridden."""
+        basePath = "http://127.0.0.1:8000/experiment/"
+        requests.post(basePath + self.mode, params={"rid": self.rid}, timeout=10)
+        self.done.emit()
+
+
 class SchedulerApp(qiwis.BaseApp):
     """App for displaying the submitted experiment queue.
 
     Attributes:
         schedulerFrame: The frame that shows the submitted experiment queue.
+        menu: The QMenu instance to display menu when right-clicked.
+        actions: The dictionary of possible outcomes made by menu.
     """
 
     def __init__(self, name: str, parent: Optional[QObject] = None):
         """Extended."""
         super().__init__(name, parent=parent)
         self.schedulerFrame = SchedulerFrame()
+        self.schedulerFrame.queueView.rightButtonPressed.connect(self.displayMenu)
+        self.menu = QMenu(self.schedulerFrame)
+        # TODO(giwon2004) Remove icon space from the menu list (use menu.setStyleSheet)
+        self.actions = {
+            "edit": self.menu.addAction("Edit"),
+            "delete": self.menu.addAction("Delete"),
+            "terminate": self.menu.addAction("Request termination")
+        }
+
+    @pyqtSlot(QMouseEvent)
+    def displayMenu(self, event: QMouseEvent):
+        """Displays the menu pop-up.
+
+        Args:
+            event: Information holder containing the clicked position.
+        """
+        for i in range(self.schedulerFrame.model.rowCount()):
+            index = self.schedulerFrame.model.index(i)
+            if self.schedulerFrame.queueView.rectForIndex(index).contains(event.pos()):
```

Comment:
Oh.. Is it the best solution?

Sure, this code seems to work well so you don't have to modify it now, but it looks little convoluted.

Reply:
True, but I could not find the function that directly gives the index :(